src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10603,6 +10615,20 @@ internal ExpressionSyntax ParseConsequenceSyntax()
                         return expr;
                 }
             }
+
+            bool isExclamationSequenceFollowedByConditionalConsequencePart()
```

Comment:
This method name is `isExclamationSequenceFollowedBy` but this method does not guarantee there is an exclamation token. It will return true if the current token is an open paren for example with no exclamation.

Reply:
0 or more exclamations 😉

Reply:
Perhaps "isOptionalExclamationSequence..."

---
In reply to: [489068851](https://github.com/dotnet/roslyn/pull/47712#discussion_r489068851) [](ancestors = 489068851)